### DISCUSSION TITLE
Is it possible to parametrize Steps template?

### DISCUSSION THREAD

**Discussion by Adiqq** [argoproj/argo-workflows#11455] [Q&A] [answered]
Hi,

Let's say that I have WorkflowTemplateA and WorkflowTemplateB.
In WorkflowTemplateA, I defined template with name: steps, which has some steps, which are some some templates with input parameters.
In WorkflowTemplateB, I defined some template that is entrypoint and has steps. It contains step that is `workflowTemplateRef` to template with name: steps, that was defined in WorkflowTemplateA. WorkflowTemplateB accepts workflow parameters and I would like to pass them as input parameters for this workflowTemplateRef.

What I want to achieve is to have small templates that do single thing, but I would like to compose them into reusable, parameterizable steps templates that can be used from steps template that is entrypoint. However it doesn't seems to work and I can't find any example of using Steps input parameters (is it supported use case?).

**Comment by agilgur5** [comment, marked as answer]
This should be doable with [`templateRef`](https://argoproj.github.io/argo-workflows/fields/#templateref), which can be used inside of a step.

Have you tried that?
